executor/aggfuncs/func_first_row.go

```diff
@@ -475,25 +500,25 @@ func (e *firstRow4Enum) ResetPartialResult(pr PartialResult) {
 func (e *firstRow4Enum) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup []chunk.Row, pr PartialResult) (memDelta int64, err error) {
 	p := (*partialResult4FirstRowEnum)(pr)
 	if p.gotFirstRow {
-		return 0, nil
+		return memDelta, nil
 	}
 	for _, row := range rowsInGroup {
 		d, err := e.args[0].Eval(row)
 		if err != nil {
-			return 0, err
+			return memDelta, err
 		}
 		p.gotFirstRow, p.isNull, p.val = true, d.IsNull(), d.GetMysqlEnum().Copy()
```

Comment:
The `Enum.Name` is a string,  the `Enum.Copy()` will use the `stringutil.Copy()` to deep copy a new string,  so we should evaluate the memory usage of the new string here.